fe/fe-core/src/main/java/com/starrocks/http/rest/ExecuteSqlAction.java

```diff
@@ -0,0 +1,115 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.http.rest;
+
+/* Usage:
+   eg:
+        POST  /api/sql  post_data=query
+ return:
+        [{rowObj},{rowObj},...]
+ */
+
+import com.google.common.base.Strings;
+import com.starrocks.analysis.QueryStmt;
+import com.starrocks.analysis.ShowStmt;
+import com.starrocks.analysis.SqlParser;
+import com.starrocks.analysis.SqlScanner;
+import com.starrocks.analysis.StatementBase;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.util.SqlParserUtils;
+import com.starrocks.http.ActionController;
+import com.starrocks.http.BaseRequest;
+import com.starrocks.http.BaseResponse;
+import com.starrocks.http.IllegalArgException;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.OriginStatement;
+import com.starrocks.qe.StmtExecutor;
+import com.starrocks.sql.ast.QueryStatement;
+import io.netty.handler.codec.http.HttpMethod;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.StringReader;
+import java.util.List;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
+import static io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR;
+
+public class ExecuteSqlAction extends RestBaseAction {
+    private static final Logger LOG = LogManager.getLogger(ExecuteSqlAction.class);
+
+    public ExecuteSqlAction(ActionController controller) {
+        super(controller);
+    }
+
+    public static void registerAction(ActionController controller) throws IllegalArgException {
+        controller.registerHandler(HttpMethod.POST, "/api/sql", new ExecuteSqlAction(controller));
+    }
+
+    @Override
+    protected void executeWithoutPassword(BaseRequest request, BaseResponse response) throws DdlException {
```

Comment:
we should have the authentication and authentication

Reply:
Did you means privilege check? PrivilegeChecker added now.